### PR TITLE
docs: update readme

### LINKED ISSUE #90: Make clearer to use the factory

### PR DESCRIPTION
**Pull request by Jean-Grimal**
Fixes #90 and #92

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,23 +18,15 @@ This oracle handles the following cases among others (let's say that our pair is
...
-A specific implementation, the `WstEthStEthExchangeRateChainlinkAdapter`, provides the exchange rate between wstETH and stETH as a Chainlink-interface-compliant feed.
-
-This adapter is deployed on Ethereum Mainnet at the address [0x905b7dAbCD3Ce6B792D874e303D336424Cdb1421](https://etherscan.io/address/0x905b7dabcd3ce6b792d874e303d336424cdb1421#code).
-
-## MorphoChainlinkOracleV2Factory
+### MorphoChainlinkOracleV2Factory
 
 The `MorphoChainlinkOracleV2Factory` is a factory that deploys `MorphoChainlinkOracleV2` instances.
 
 The factory facilitates the creation and indexing of `MorphoChainlinkOracleV2` oracles. This factory contract streamlines the process of deploying new oracles, catering to various market pairs without the need to deploy individual contracts manually for each pair.
```
Comment (MerlinEgalite) [inline README.md:25]: i think you can delete this part which is useless

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -18,23 +18,11 @@ This oracle handles the following cases among others (let's say that our pair is
 - A/C and C/B are feeds (typically, WBTC/BTC and BTC/ETH).
 - A'/C and B/C are feeds, and there is an exchange rate between A and A'. (typically A=sDAI and A'=DAI).
 
-## WstETH/stETH Exchange Rate Adapter
-
-A specific implementation, the `WstEthStEthExchangeRateChainlinkAdapter`, provides the exchange rate between wstETH and stETH as a Chainlink-interface-compliant feed.
-
-This adapter is deployed on Ethereum Mainnet at the address [0x905b7dAbCD3Ce6B792D874e303D336424Cdb1421](https://etherscan.io/address/0x905b7dabcd3ce6b792d874e303d336424cdb1421#code).
+### MorphoChainlinkOracleV2Factory
```
Comment (QGarchery) [inline README.md:21]:
```suggestion
```
otherwise we have an empty title

File: README.md [hunk 3]
```diff
@@ -18,23 +18,11 @@ This oracle handles the following cases among others (let's say that our pair is
...
-
-The factory facilitates the creation and indexing of `MorphoChainlinkOracleV2` oracles. This factory contract streamlines the process of deploying new oracles, catering to various market pairs without the need to deploy individual contracts manually for each pair.
-
-### Usage
-
-The Ethereum Mainnet address of this factory is [0x3A7bB36Ee3f3eE32A60e9f2b33c1e5f2E83ad766](https://etherscan.io/address/0x3a7bb36ee3f3ee32a60e9f2b33c1e5f2e83ad766#code).
+#### Deploy an Oracle
 
-To create a new `MorphoChainlinkOracleV2` oracle using the factory, call the `createMorphoChainlinkOracleV2` function with the following parameters:
+To deploy a `MorphoChainlinkOracleV2` on the Ethereum mainnet, use the `MorphoChainlinkOracleV2Factory` to create a new oracle. Call the `createMorphoChainlinkOracleV2` function with the following parameters:
```
Comment (QGarchery) [inline README.md:25]:
```suggestion
To deploy a `MorphoChainlinkOracleV2` on Ethereum mainnet, use the `MorphoChainlinkOracleV2Factory` to create a new oracle. Call the `createMorphoChainlinkOracleV2` function with the following parameters:
```
[optional]

Comment (QGarchery) [inline README.md:25]: Let's put every sentence in a new line. It's easier that way to see the diff when doing big changes

**Review by QGarchery — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -18,23 +18,10 @@ This oracle handles the following cases among others (let's say that our pair is
...
-
-The factory facilitates the creation and indexing of `MorphoChainlinkOracleV2` oracles. This factory contract streamlines the process of deploying new oracles, catering to various market pairs without the need to deploy individual contracts manually for each pair.
-
-### Usage
-
-The Ethereum Mainnet address of this factory is [0x3A7bB36Ee3f3eE32A60e9f2b33c1e5f2E83ad766](https://etherscan.io/address/0x3a7bb36ee3f3ee32a60e9f2b33c1e5f2e83ad766#code).
+## Deploy an Oracle
 
-To create a new `MorphoChainlinkOracleV2` oracle using the factory, call the `createMorphoChainlinkOracleV2` function with the following parameters:
+To deploy a `MorphoChainlinkOracleV2` on the Ethereum mainnet, use the `MorphoChainlinkOracleV2Factory` to create a new oracle.
```
Comment (MathisGD) [inline README.md:23]:
```suggestion
To deploy a `MorphoChainlinkOracleV2` on Ethereum, it is highly recommended to use the factory `MorphoChainlinkOracleV2Factory`.
```

File: README.md [hunk 5]
```diff
@@ -46,9 +33,14 @@ To create a new `MorphoChainlinkOracleV2` oracle using the factory, call the `cr
...
 - `salt`: A unique identifier to create deterministic addresses for deployed oracles.
 
-### Deploy an Oracle
+**Warning:** If there is an ERC4626-compliant vault for `baseVault` or `quoteVault`, the `baseTokenDecimals` or `quoteTokenDecimals` are still the decimals of the asset in the vault, and not the decimals of the Vault asset.
+E.g: for a MetaMorpho WETH vault, as `baseVault`, the `baseTokenDecimals` is 18 as WETH has 18 decimals.
 
-To deploy a `MorphoChainlinkOracleV2` on the Ethereum mainnet, use the `MorphoChainlinkOracleV2Factory` to create a new oracle.
+### Addresses
+
+The Ethereum Mainnet address of this factory is [0x3A7bB36Ee3f3eE32A60e9f2b33c1e5f2E83ad766](https://etherscan.io/address/0x3a7bb36ee3f3ee32a60e9f2b33c1e5f2e83ad766#code).
```
Comment (MathisGD) [inline README.md:41]:
```suggestion
The address on Ethereum of this factory is [0x3A7bB36Ee3f3eE32A60e9f2b33c1e5f2E83ad766](https://etherscan.io/address/0x3a7bb36ee3f3ee32a60e9f2b33c1e5f2e83ad766#code).
```

File: README.md [hunk 6]
```diff
@@ -18,23 +18,10 @@ This oracle handles the following cases among others (let's say that our pair is
...
-The factory facilitates the creation and indexing of `MorphoChainlinkOracleV2` oracles. This factory contract streamlines the process of deploying new oracles, catering to various market pairs without the need to deploy individual contracts manually for each pair.
-
-### Usage
-
-The Ethereum Mainnet address of this factory is [0x3A7bB36Ee3f3eE32A60e9f2b33c1e5f2E83ad766](https://etherscan.io/address/0x3a7bb36ee3f3ee32a60e9f2b33c1e5f2e83ad766#code).
+## Deploy an Oracle
 
-To create a new `MorphoChainlinkOracleV2` oracle using the factory, call the `createMorphoChainlinkOracleV2` function with the following parameters:
+To deploy a `MorphoChainlinkOracleV2` on the Ethereum mainnet, use the `MorphoChainlinkOracleV2Factory` to create a new oracle.
+Call the `createMorphoChainlinkOracleV2` function with the following parameters:
```
Comment (MathisGD) [inline README.md:24]:
```suggestion
To do so, call the `createMorphoChainlinkOracleV2` function with the following parameters:
```

File: README.md [hunk 7]
```diff
@@ -46,9 +33,14 @@ To create a new `MorphoChainlinkOracleV2` oracle using the factory, call the `cr
 - `quoteTokenDecimals`: Decimal precision of the quote asset.
 - `salt`: A unique identifier to create deterministic addresses for deployed oracles.
 
-### Deploy an Oracle
+**Warning:** If there is an ERC4626-compliant vault for `baseVault` or `quoteVault`, the `baseTokenDecimals` or `quoteTokenDecimals` are still the decimals of the asset in the vault, and not the decimals of the Vault asset.
```
Comment (MathisGD) [inline README.md:36]:
```suggestion
**Warning:** If there is an ERC4626-compliant vault for `baseVault` or `quoteVault`, the `baseTokenDecimals` or `quoteTokenDecimals` are still the decimals of the underlying asset of the vault, and not the decimals of the Vault itself.
```

File: README.md [hunk 8]
```diff
@@ -46,9 +33,14 @@ To create a new `MorphoChainlinkOracleV2` oracle using the factory, call the `cr
...
+E.g: for a MetaMorpho WETH vault, as `baseVault`, the `baseTokenDecimals` is 18 as WETH has 18 decimals.
```
Comment (MathisGD) [inline README.md:37]: metamorpho also has 18 decimals (and most vaults do), so the example is not perfect

to me we can just remove it

**Comment by MerlinEgalite** [comment]
Apply changes and merged, PR pending for too long

**Pull request merged**